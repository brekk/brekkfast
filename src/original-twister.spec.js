import Twister from "./original-twister"

const SEED = {
  ONE: 100,
  TWO: 200,
}

function testWithSeed(seed) {
  describe(`Twister with seed: ${seed}`, () => {
    let TWIST
    beforeAll(() => {
      TWIST = new Twister(seed)
    })

    test("integer", () => {
      const x = TWIST.random_int()
      expect(x).toMatchSnapshot()
    })
    test("integer31", () => {
      const x = TWIST.random_int31()
      expect(x).toMatchSnapshot()
    })
    test("inclusive", () => {
      const x = TWIST.random_incl()
      expect(x).toMatchSnapshot()
    })
    test("random", () => {
      const x = TWIST.random()
      expect(x).toMatchSnapshot()
    })
    test("exclusive", () => {
      const x = TWIST.random_excl()
      expect(x).toMatchSnapshot()
    })
    test("long", () => {
      const x = TWIST.random_long()
      expect(x).toMatchSnapshot()
    })
  })
}

testWithSeed(SEED.ONE)
testWithSeed(SEED.TWO)
