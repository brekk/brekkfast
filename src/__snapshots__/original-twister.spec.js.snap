// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twister with seed: 100 exclusive 1`] = `0.42451759229879826`;

exports[`Twister with seed: 100 inclusive 1`] = `0.2783693827871162`;

exports[`Twister with seed: 100 integer 1`] = `2333906440`;

exports[`Twister with seed: 100 integer31 1`] = `1441295756`;

exports[`Twister with seed: 100 long 1`] = `0.5263824674842963`;

exports[`Twister with seed: 100 random 1`] = `0.4120463968720287`;

exports[`Twister with seed: 200 exclusive 1`] = `0.5944201449165121`;

exports[`Twister with seed: 200 inclusive 1`] = `0.22654742380290932`;

exports[`Twister with seed: 200 integer 1`] = `4070049562`;

exports[`Twister with seed: 200 integer31 1`] = `1061165364`;

exports[`Twister with seed: 200 long 1`] = `0.25760976544252545`;

exports[`Twister with seed: 200 random 1`] = `0.1485590497031808`;
